src/__tests__/Exame.test.ts

```diff
@@ -0,0 +1,38 @@
+import request from 'supertest';
+import { Connection, getConnection } from 'typeorm';
+import { app } from '../app';
+
+import createConnection from '../database';
+let connection: Connection;
+
+describe("exames", () => {
+    beforeAll(async () => {
+        connection = await createConnection();
+        await connection.runMigrations();
+    })
+
+    afterAll(async () => {
+        connection.undoLastMigration();
+        await connection.close();
+    })
+
+    it("Should be able to create a new Exame", async () => {
+        const response = await request(app).post("/exames").send({
+            nome: "teste",
+            autorizacao: true
+        });
+
+        expect(response.status).toBe(201);
+        expect(response.body).toHaveProperty("id");
+    })
+
+    it("Should not be able to create a new Pacient if it already exists", async () => {
```

Comment:
Se for passado só "nome" ou só "autorizacao" o que deve acontecer? Acho importante ter esses testes também

Reply:
Nesse ponto do POST pra o banco imaginei que isso já teria sido tratado no front, sendo passado já todos os dados para o banco na hora do envio.

Reply:
Mas há outros casos que poderíamos considerar, por exemplo o formato do CPF. Talvez fosse interessante verificar se o formato informado está correto, mas como não teremos nenhuma busca que precise garantir que sejam o cpf tenha um formato exato, então a gente pode flexibilizar

Reply:
Mas claro, testar nunca é de mais, sempre vai haver algo a ser verificado

Reply:
> Mas há outros casos que poderíamos considerar, por exemplo o formato do CPF. Talvez fosse interessante verificar se o formato informado está correto, mas como não teremos nenhuma busca que precise garantir que o cpf tenha um formato exato, então a gente pode flexibilizar

Nesses casos isso tbm não é verificado antes de enviar não? O meu pensamento é que o back tem esse papel de apenas armazenar e servir as informações com autênticações etc, então não precisa ser validado o formato do dado, e sim se ele já existe ou não na base de dados

Reply:
Tem a questão da Regra de Negócio, que é um ponto que o back deve tratar também. Inclusive, é uma das funções do controller

Reply:
Mas há casos que é feita dupla verificação: tanto no back quanto no front. No entanto, no caso de uma API, acho que é interessante adicionar o máximo de verificações possível, tento em vista que a ideia é servir diferentes aplicações. Nesse sentido, acho que é importante não confiar apenas que o front vai fazer a verificação

Reply:
Certo. Então você tá sugerindo adicionar a verificação tanto aqui em Exame quanto em Paciente também?

Reply:
Por hora faça e exame. Eu to fazendo uma tarefa de Pacientes, então posso ajustar lá

Reply:
Ok.